fix: Update rate-limit websocket docs (#2609)
nestjs/docs.nestjs.com: content/security/rate-limiting.md

```typescript
@Injectable()
export class WsThrottlerGuard extends ThrottlerGuard {
  async handleRequest(context: ExecutionContext, limit: number, ttl: number): Promise<boolean> {
    const client = context.switchToWs().getClient();
    const ip = client._socket.remoteAddress
    const key = this.generateKey(context, ip);
    const { totalHits } = await this.storageService.increment(key, ttl);

    if (totalHits > limit) {
      throw new ThrottlerException();
    }

    return true;
  }
}
```
> info **Hint** If you using ws, it is necessary to replace the `_socket` with `conn`

> There are some things to take keep in mind when working with websockets:

Reviewer: ```suggestion
There's a few things to keep in mind when working with WebSockets:
```

> - You cannot bind the guard with `APP_GUARD` or `app.useGlobalGuards()` due to how Nest binds global guards.

Reviewer: ```suggestion
- Guard cannot be registered with the `APP_GUARD` or `app.useGlobalGuards()`
```

> - When a limit is reached, Nest will emit an `exception` event, so make sure there is a listener ready for this.

Reviewer: ```suggestion
- When a limit is reached, Nest will emit an `exception` event, so make sure there is a listener ready for this
```

```typescript
@Injectable()
export class WsThrottlerGuard extends ThrottlerGuard {
  async handleRequest(context: ExecutionContext, limit: number, ttl: number): Promise<boolean> {
    const client = context.switchToWs().getClient();
    const ip = ['_socket']
      .map((key) => client[key])
      .filter((obj) => obj)

>       .shift().remoteAddress;

Reviewer 2: This entire map and filter is not longer needed, right? just `client._socket.remoteAddress`

Author: Yes

```typescript
@Injectable()
export class WsThrottlerGuard extends ThrottlerGuard {
  async handleRequest(context: ExecutionContext, limit: number, ttl: number): Promise<boolean> {
    const client = context.switchToWs().getClient();
    // this is a generic method to switch between `ws` and `socket.io`. You can choose what is appropriate for you
    const ip = ['conn', '_socket']
      .map((key) => client[key])
      .filter((obj) => obj)

>       .shift().remoteAddress;

Reviewer 2: Rather than make this generic method, could we have the example be for `socket.io` and then in an `info` block underneath mention that for `ws` `_socket` should be used instead of `conn`? We don't do something like the above anywhere else in the docs (that I'm aware of) and it would lead to misunderstanding of what's happening here

Author: Sure
